api-server/src/main/java/vermillion/http/HttpServerVerticle.java

```diff
@@ -44,1048 +44,1139 @@
+
+										// Getting common directory for all ids
+										String authorisedIdPrefix = commonPrefix(authorisedIds);
+										int lastDirIndex = authorisedIdPrefix.lastIndexOf('/');
+										if (lastDirIndex != -1) {
+										 CONSUMER_PATH += authorisedIdPrefix.substring(0, lastDirIndex + 1); 
+										}
+									}
+								}
+								
+								return Completable.complete();
+							})
+					.subscribe(
+							() -> context.reroute(CONSUMER_PATH), t -> apiFailure(context, t));
+		} else {
+			checkAuthorisation(token, READ_SCOPE, requestedIds)
+					.andThen(
+							Completable.fromCallable(
+									() -> {
+
+										logger.debug("Requested IDs = " + requestedIds.encode());
+										for (int i = 0; i < requestedIds.size(); i++) {
+											logger.debug("File=" + basePath + requestedIds.getString(i));
+											if (Files.notExists(Paths.get(basePath + requestedIds.getString(i)))) {
+												return Completable.error(
+														new UnauthorisedThrowable("Requested resource ID(s) is not present"));
+											}
+										}
+
+										for (int i = 0; i < requestedIds.size(); i++) {
+											String resourceId = requestedIds.getString(i);
+											String consumerResourceDir = WEBROOT + "consumer/" + token + "/"
+															+ resourceId.substring(0,resourceId.lastIndexOf('/'));
+
+											// Create consumer directory path if it does not exist
+											new File(consumerResourceDir).mkdirs();
+											logger.debug("Created consumer subdirectory");
+
+											Path consumerResourcePath =
+													Paths.get(WEBROOT + "consumer/" + token + "/" + resourceId);
+											Path providerResourcePath = Paths.get(basePath + resourceId);
+
+											// TODO: This could take a very long time for multiple large files
+											try {
+												Files.createSymbolicLink(consumerResourcePath, providerResourcePath);
+											} catch (FileAlreadyExistsException ignored) {
+										 
+					} catch (Exception e) {
+												return Completable.error(
+														new InternalErrorThrowable("Could not create symlinks"));
+											}
+										}
+										
+										// Appending Consumer Path
+										if(!requestedIds.isEmpty()){
+											if(requestedIds.size() == 1){
+												CONSUMER_PATH += requestedIds.getString(0) + "/";
+											}
+											else {
+												String requestedIdPrefix = commonPrefix(requestedIds);
+												int lastDirIndex = requestedIdPrefix.lastIndexOf('/');
+												if (lastDirIndex != -1) {
+													CONSUMER_PATH += requestedIdPrefix.substring(0, lastDirIndex + 1);
+												}
+											}
+										}
+
+										return Completable.complete();
+									}))
+					.subscribe(
+							() -> context.reroute(CONSUMER_PATH), t -> apiFailure(context, t));
+		}
+	}
+
+	// Publish API for timeseries data as well as static files
+	public void publish(RoutingContext context) {
+		// TODO: Check scope before allowing publish - very imp
```

Comment:
Remove todos when they are implemented. Close the corresponding issue by linking the commit.